src/wp-includes/rest-api.php

```diff
@@ -295,6 +295,25 @@ function rest_api_loaded() {
 	 */
 	define( 'REST_REQUEST', true );
 
+	// Populate the correct $_SERVER variables via an alternate header for fastcgi compatibility.
```

Comment:
I'm a little unsure about putting this code here in `rest_api_loaded` which seems primarily concerned with bootstrapping `$server` so as to handle an incoming request.

Is there a better place that this code could live? One idea: Maybe split it off into a separate function called something like `rest_api_basic_authentication` which is invoked via a `add_action( 'parse_request', 'rest_api_basic_authentication', 9 );` in `default-filters.php`?

Reply:
My gut inclination is to avoid cluttering up global hook registries with stuff that's only ever useful for rest api stuff, so I'd kinda rather keep it nested down here, but that could be more a me-issue not a real objection.

Reply:
I tend to agree with @georgestephanis here. This is not useful in any other place and it could be that we don't want those `$_SERVER` properties to exist in any other context.